### ISSUE TITLE
Admin Upload API throws an axios error

### ISSUE THREAD

**Issue by renaudAmsellem** [medusajs/medusa#9241]
### Describe the bug

Hi, first of all, thanks for the lib! I'm having a great time building with it.

I'm getting an error when trying to upload an image with the admin API with the default file system: 

```js
await medusa.admin.uploads.create(filePath)
```

The error: `Error: Data after transformation must be a string, an ArrayBuffer, a Buffer, or a Stream`

### System information

Medusa version (including plugins): 
    "@medusajs/file-local": "1.0.4",
    "@medusajs/medusa": "^1.20.9",
Node.js version: v20.17.0
Database: 16.4
Operating system: mac os sonoma 14.1.1

### Steps to reproduce the behavior

I've created a repo (first commit is "medusa new", second and third commit are actual code)
1. Go to https://github.com/renaudAmsellem/issue-upload-image
2. execute script `node scripts/upload-image.mjs`
3. View error

### Expected behavior

The file is uploaded correctly to the file service configured. In this case, to the `uploads` directory.
The API returns the URL of the file as per the doc.

### Additional context

I saw that the error comes from this file : In node_modules/@medusajs/medusa-js/node_modules/axios/lib/adapters/http.js:

```js
if (data && !utils.isStream(data)) {
      if (Buffer.isBuffer(data)) {
        // Nothing to do...
      } else if (utils.isArrayBuffer(data)) {
        data = Buffer.from(new Uint8Array(data));
      } else if (utils.isString(data)) {
        data = Buffer.from(data, 'utf-8');
      } else {
        return reject(createError(
          'Data after transformation must be a string, an ArrayBuffer, a Buffer, or a Stream',
          config
        ));
      }

      // Add Content-Length header if data exists
      if (!headerNames['content-length']) {
        headers['Content-Length'] = data.length;
      }
    }
```

And when i log `data`: `FormData { files: '[object FormData]' }`

i've tried to transform the file into an ArrayBuffer, a Stream. But I always get the same error.
I've described the error on [discord](https://discord.com/channels/876835651130097704/1270058134945398846) but no luck there.

**Comment by olivermrbl** [comment]
We'll look into this.

Out of curiosity, is there a reason you are not using Medusa 2.0? Medusa 1.0 has moved into maintenance mode and is only patched with critical fixes.

**Comment by renaudAmsellem** [comment]
Thanks.

Well I was under the feeling medusa v2 was still under development and that a stable release was not out yet.
Would you advice me to go to the 2.0 even if it's not on stable yet?

**Comment by olivermrbl** [comment]
@renaudAmsellem, we expect to publish the official release of 2.0 end of October. If your launch-date is not within the next month, I would go with 2.0 :)

**Comment by abdullah-afzal** [comment]
@olivermrbl ,  Can we expect the migration guide from V1 to V2 with the official release as well?

**Comment by olivermrbl** [comment]
No, not alongside the release. We expect to publish a migration guide within the two months following it.
